tests/src/gui/testqgscompoundcolorwidget.cpp

```diff
@@ -88,5 +90,38 @@ void TestQgsCompoundColorWidget::testCmykConversion()
   QCOMPARE( w.color(), QColor::fromCmyk( 170, 85, 0, 225, 50 ) );
 }
 
+void TestQgsCompoundColorWidget::testComponentChange()
+{
+  QgsSettings().setValue( QStringLiteral( "Windows/ColorDialog/activeComponent" ), 3 );
```

Comment:
Since you're modifying settings, can you add this logic to the start of the test: https://github.com/qgis/QGIS/blob/master/tests/src/core/testqgsapplication.cpp#L58-L61

Reply:
Done